esmvaltool/utils/cmorizers/obs/utilities.py

```diff
@@ -49,19 +51,20 @@ def _fix_dim_coordnames(cube):
         cube.coord(axis='Y').long_name = 'latitude coordinate'
         cube.coord(axis='Y').units = Unit('degrees')
 
-    if cube.coord(axis='Z') in cube.coords():
-        if cube.coord(axis='Z').var_name == 'depth':
-            cube.coord(axis='Z').standard_name = 'depth'
-            cube.coord(axis='Z').long_name = 'ocean depth coordinate'
-            cube.coord(axis='Z').var_name = 'lev'
-        if cube.coord(axis='Z').var_name == 'pressure':
-            cube.coord(axis='Z').standard_name = 'air_pressure'
-            cube.coord(axis='Z').long_name = 'pressure'
-            cube.coord(axis='Z').var_name = 'air_pressure'
```

Comment:
please don't remove and reformat this bit, there is a good reason why the check is done on the (cmor-standard) Z axis and not on the number of axes: it could well be that the cube has two (or even one axis) and Z is amongst them

Reply:
Here is the problem:
If you deal with a 2D field (time,lon,lat) the cube inquire `cube.coord(axis='Z').var_name` generate an error and the code stops, but the dimension simply does not exist. 
I put a comment along with the check on the number of dimensions as I also agree that it is not the best way to do it ...
Do you think there is an alternative way to avoid the code stopping when dealing with 2D fields?

Reply:
in the future please mark your comments with `# TODO` - this way the markup will highlight it and Codacy will pick it up and let you know about it. This is an ommission on my side - intially I assumed that the cmoriation will be done on data that has some basic cmorization (CF standard data in iris automatically assigns x,y,z and t axes to the dim coords). I need to put in a guesser for totally non-cmor data using `iris.utils.guess_coord_axis()` - or even better this can be done in the cmorization script - have a look at that function and use it in your script; for the z-axis to be correctly set you will need to set the positive attribute to the axis eg `coord.attributes['positive'] = 'up`'

Reply:
I guess adding a 3rd dimension to comply with the CF standard data in iris (x,y,z,t) will lead to a problem when checking the variable rank against CMOR standard, which is **XY-na** in this particular case.
I feel like suggesting a mid way solution  ... If we move `_fix_dim_coordnames` out of `_fix_coords` in `utilities.py` then I can redefine a local version of it within the cmorizer script. If I'm right, It comes at the cost of updating its usage only in cmorize_obs_WOA.py.
Does it works for You?